DigitalLearningSolutions.Web/Views/TrackingSystem/Delegates/DelegateProgress/EditSupervisor.cshtml

```diff
@@ -61,9 +63,18 @@
     </form>
 
     @if (Model.AccessedVia.Equals(DelegateAccessRoute.CourseDelegates)) {
-      <vc:cancel-link asp-controller="DelegateProgress" asp-action="Index" asp-all-route-data="@routeParamsForBackLink" />
+      <vc:cancel-link-with-return-page-query
+        asp-controller="DelegateProgress"
+        asp-action="Index"
+        route-data="@routeParamsForBackLink"
+        return-page-query="@Model.ReturnPageQuery" />

```

Comment:
Couple of things for the modifications to the links of these Edit pages. 

Firstly, the DelegateProgress page doesn't have anywhere that it can use the return page query stuff for, as it does not have any cards. I also don't think that it can pass back the return page query to the view delegates page as the only link backwards would be the breadcrumbs, which we aren't preserving the return page query for. So I'm not sure on the usefulness of this.

Secondly, this has been applied to the CourseDelegates version of the cancel link, but those pages do not pass in the return page query. As such, this page will fail to load when accessing from CourseDelegates/DelegateProgress. Admittedly, we are going to be changing those pages around in the big merge ticket, but it might be nice to not break the functionality until that is complete. 

So, overall, we can probably keep this as it was.